src/main/java/mediator/Mediator.java

```diff
@@ -30,268 +30,429 @@
+	 * @param carrierFrequencies The frequency band that will be used with this carrier.
+	 * @param transmittingPower  The transmitting power for this carrier.
+	 * @param ratType            The RAT type of the radio unit that this carrier will be associated with.
+	 * @return The created Carrier based on the RAT type of the radio unit it will be associated with.
+	 */
+	private synchronized Carrier createCarrier(List<RfPort> rfPorts, FrequencyBand carrierFrequencies, Double transmittingPower, RatType ratType) {
+		switch (ratType) {
+		case WCDMA:
+		{
+			try {
+				return carrierManagement.createWcdmaCarrier(rfPorts, carrierFrequencies, transmittingPower);
+			} catch (ArrayIndexOutOfBoundsException ex) {
+				return null;
+			}
+		}
+		case LTE:
+		{
+			try {
+				return carrierManagement.createLteCarrier(rfPorts, carrierFrequencies, transmittingPower);
+			} catch (ArrayIndexOutOfBoundsException ex) {
+				return null;
+			}
+		}
+		default:
+		{
+			System.out.print("[ERROR] Invalid RAT type detected, failed to create a carrier.");
+			return null;
+		}
+		}
+	}
+
+	/**
+	 * List any registered radio units that contain the parameter being passed.
+	 *
+	 * @param param The parameter that will be used to query for radio units. Examples
+	 *              include a RatType, a RadioUnitState, a FrequencyBand, or a String name.
+	 */
+	private void listRuByParam(Object param) {
+		if (param instanceof RatType) {
+			List<ManagedRadioUnit> radioUnits = radioUnitRegistry.getByRatType((RatType) param);
+			radioUnits.forEach(System.out::println);
+		} else if (param instanceof RadioUnitStateE) {
+			List<ManagedRadioUnit> radioUnits = radioUnitRegistry.getByState((RadioUnitStateE) param);
+			radioUnits.forEach(System.out::println);
+		} else if (param instanceof FrequencyBand) {
+			List<ManagedRadioUnit> radioUnits = radioUnitRegistry.getByBand((FrequencyBand) param);
+			radioUnits.forEach(System.out::println);
+		} else if (param instanceof String) {
+			List<ManagedRadioUnit> radioUnits = radioUnitRegistry.getByName((String) param);
+			radioUnits.forEach(System.out::println);
+		}
+	}
+
+	/**
+	 * Notify listeners of the RAT Type for a specific radio unit.
+	 *
+	 * @param evt The event that contains the necessary information to return alongside the RAT Type.
+	 */
+	@SuppressWarnings("unchecked")
+	private void messageRatType(PropertyChangeEvent evt) {
+		List<Object> params = ((ArrayList<Object>) evt.getNewValue());
+		ManagedRadioUnit radio = getRadioUnit((String) params.get(0));
+		if (radio == null)
+		{
+			System.out.println("Unable to message RAT type listeners: Radio Unit not found!");
+			return;
+		}
+
+		List<Object> newParams = new ArrayList<>(Arrays.asList(params.get(0), radio.getRatType()));
+		support.firePropertyChange(evt.getPropertyName(), params.get(1), newParams);
+	}
+
+	/**
+	 * Print all alarms currently raised in the network.
+	 */
+	private void printNetworkAlarms() {

```

Comment:
Yeah, my initial thought with this method was including the assumption that "NO_ALARM" was an alarm and I am just now realizing that makes zero sense. Thanks for fixing this.